pkgs/development/tools/misc/tie/default.nix

```
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
   };
 
   buildPhase = ''
-    cc tie.c -o tie
+    ${stdenv.cc.targetPrefix}cc tie.c -O2 -o tie
```

```suggestion
    $CC tie.c -o tie
```
`-O2` is already implied by the default hardening flags.
The use of `$CC` is a matter of personal taste.

---

`-O2` is already used even though this is a manual call to `cc`?

---

it's added by cc-wrapper, you can check with `export NIX_DEBUG=1` before.